### PR TITLE
Introduce test-suite directory

### PR DESCRIPTION
**Pull request by popzxc**
## Overview

This PR introduces a new directory "test_suite" which (already) contains `testkit` and (will) contain all the integration tests later (starting from python).

### CODE REVIEW

**Comment by popzxc** [comment]
Just how?
https://codecov.io/gh/exonum/exonum/pull/1515/changes

**Comment by slowli** [comment]
[Very important question](https://en.wikipedia.org/wiki/Law_of_triviality): should it be `test_suite` or `test-suite`? We use `kebab-case` in some other directory names (e.g., `proto-derive`, `cryptocurrency-advanced`; although `sample_runtime` is just nearby), and IMO it's more Ａ Ｅ Ｓ Ｔ Ｈ Ｅ Ｔ Ｉ Ｃ.

**Comment by popzxc** [comment]
Well, personally I usually try to use a Python convention:
- [Project names are in kebab-case](https://github.com/search?l=Python&q=stars%3A%3E100&type=Repositories)
- [Module names are in snake case](https://www.python.org/dev/peps/pep-0008/#package-and-module-names)

That explains why `test_suite` (aka test module of Exonum project), but `exonum-py-test` (aka standalone installable Python package).
Inconsistent? Well, we have a monorepo `¯\_(ツ)_/¯`

**Comment by aleksuss** [comment]
@popzxc But we can interpret a `test-suite` like a project which contains tests for exonum framework, can't we? I'd prefer `kebab-case` here. And `sample_runtime` I'd rename as well. Just have skipped it. Or. If we will chose `snake-case` than we should rename all directories which are in `kebab-case`. But use only one style for directories.

**Comment by popzxc** [comment]
Well, seems reasonable.

**Comment by popzxc** [comment]
Done. Lets leave `sample_runtime` as-is for now, I'll rename it in a separate PR.

File: .travis.yml [hunk 1]
```diff
@@ -146,7 +146,7 @@ jobs:
   - name: integration-tests
     install:
     - nvm install 8 && nvm use 8
-    - cd $TRAVIS_BUILD_DIR/testkit/server && npm install && cd $TRAVIS_BUILD_DIR
+    - cd $TRAVIS_BUILD_DIR/test_suite/testkit/server && npm install && cd $TRAVIS_BUILD_DIR
```
Comment (aleksuss) [inline .travis.yml]:
```suggestion
    - cd $TRAVIS_BUILD_DIR/test-suite/testkit/server && npm install && cd $TRAVIS_BUILD_DIR
```

File: README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ infrastructure providers.
 This is the main Exonum repository that includes
 
 * [Exonum core library](exonum/README.md).
-* [Exonum testing framework](testkit/README.md).
+* [Exonum testing framework](test_suite/testkit/README.md).
```
Comment (aleksuss) [inline README.md]:
```suggestion
* [Exonum testing framework](test-suite/testkit/README.md).
```

**Pull request merged**